platform/linux-generic/arch/aarch64/odp/api/abi/ticketlock_inlines.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2016-2018, Linaro Limited
+ * Copyright (c) 2021, Nokia
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @file
+ *
+ * ODP Ticketlock inline functions
+ */
+#ifndef ODP_API_ABI_TICKETLOCK_INLINES_H_
+#define ODP_API_ABI_TICKETLOCK_INLINES_H_
+
+static inline uint32_t monitorll32(uint32_t *var, int mm)
+{
+	uint32_t old;
+
+	if (mm == __ATOMIC_ACQUIRE)
+		__asm__ volatile("ldaxr %w0, [%1]"
+				 : "=&r" (old)
+				 : "r" (var)
+				 : "memory");
+	else if (mm == __ATOMIC_RELAXED)
+		__asm__ volatile("ldxr %w0, [%1]"
+				 : "=&r" (old)
+				 : "r" (var)
+				 : );
+	return old;
+}
+
+static inline void
+_odp_wait_until_equal_32(uint32_t expected, uint32_t *addr, int memorder)
```

Comment:
I'd suggest to change into _odp_wait_until_equal_acq_u32(odp_atomic_u32_t *addr, uint32_t expected)

- acquire version. we can add relaxed versions later when needed.
- u32 == unsigned 32 bit
- atomic as pointer type
- pointer as first argument